pandas/io/html.py

```diff
@@ -912,11 +909,9 @@ def _parse(flavor, io, match, attrs, encoding, displayed_only, **kwargs):
                     "different flavor.".format(flav)
                 )
 
-            retained = caught
+            raise
```

Comment:
does this change the current behavior?  `for/else` is a pretty obscure idiom that i dont have intuition for

Reply:
Also if you're inclined to tighten the `except Exception` above while in the neighborhood, that'd be cool

Reply:
The `else` clause in `for...else` would only get executed if there was no `break` in the `for` block. In this cause it would `break` if no exception was raised in the try...else

Come to think of it we could get rid of the `try...else` now too. I'll see what that looks like after this CI run completes (mostly want to see if it helps with the HTML issue visibility)

Reply:
Actually misread that. This goes through a fallback pattern of the various parsers trying one after the next. I erroneously just raised on first available - might not be well tested...

For now just reverted; can open a follow up for clean up and except Exception handling as you mentioned